### ISSUE TITLE
no need to update cache every time in region heartbeat 

### ISSUE THREAD

**Issue by siddontang** [pingcap/pd#471]
In `(c *clusterInfo) handleRegionHeartbeat` 

```go
	// Region meta is the same, update cache only.
	c.regions.setRegion(region)
```

But now `setRegion` will do:

```go
func (r *regionsInfo) setRegion(region *regionInfo) {
	if origin, ok := r.regions[region.GetId()]; ok {
		r.removeRegion(origin)
	}
	r.addRegion(region)
}
```

If the region is not changed, I think we don't need to remove first and then re-add again.

**Comment by siddontang** [comment]
@disksing 

seem we can close this now?

**Comment by disksing** [comment]
Fixed since #575.
